### PR TITLE
networkd: Tunnel allows tunnel traffic on ip6tnl devices

### PR DESCRIPTION
**Pull request by ssahani**
where the remote endpoint is a local host address.

### CODE REVIEW

File: src/network/netdev/netdev-gperf.gperf [hunk 1]
```diff
@@ -63,6 +63,7 @@ Tunnel.IPv6FlowLabel,        config_parse_ipv6_flowlabel,          0,
 Tunnel.CopyDSCP,             config_parse_bool,                    0,                             offsetof(Tunnel, copy_dscp)
 Tunnel.EncapsulationLimit,   config_parse_encap_limit,             0,                             offsetof(Tunnel, encap_limit)
 Tunnel.Independent,          config_parse_bool,                    0,                             offsetof(Tunnel, independent)
+Tunnel.AllowLocalRemote,     config_parse_tristate,                0,                             offsetof(Tunnel, allow_localremote)
```
Comment (yuwata) [inline src/network/netdev/netdev-gperf.gperf:66]: Why do you set this tristate instead of boolean?

File: src/network/netdev/tunnel.c [hunk 2]
```diff
@@ -336,6 +337,11 @@ static int netdev_ip6tnl_fill_message_create(NetDev *netdev, Link *link, sd_netl
         if (t->copy_dscp)
                 t->flags |= IP6_TNL_F_RCV_DSCP_COPY;
 
+        if (t->allow_localremote)
```
Comment (yuwata) [inline src/network/netdev/tunnel.c]: `t->allow_localremote > 0` ?

File: src/network/netdev/netdev-gperf.gperf (hunk 1 above)
Comment (ssahani) [reply src/network/netdev/netdev-gperf.gperf:66]: Let the kernel state like that we won't touch it .

File: src/network/netdev/tunnel.c (hunk 2 above)
Comment (ssahani) [reply src/network/netdev/tunnel.c]: yes

**Comment by ssahani** [comment]
Thanks @yuwata . Updated.

File: man/systemd.netdev.xml [hunk 3]
```diff
@@ -889,6 +889,14 @@
           </para>
         </listitem>
       </varlistentry>
+      <varlistentry>
+        <term><varname>AllowLocalRemote=</varname></term>
+        <listitem>
+          <para>A boolean. When true allows tunnel traffic on ip6tnl devices where the remote endpoint is a local host address.
```
Comment (yuwata) [inline man/systemd.netdev.xml]: Please use varname tag, that is, `<varname>ip6tnl</varname>`.

**Comment by ssahani** [comment]
updated

**Pull request merged**